src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2693 @@
+            occKey <- cParams ^? ProtoFields.onChainCommitmentKey
+            createGlobalContext genString bpGens occKey
+
+getCryptographicParametersV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult (Maybe CryptographicParameters))
+getCryptographicParametersV2 bHash = withUnaryCoreV2 (callV2 @"getCryptographicParameters") msg ((fmap . fmap) fromProto)
+  where
+    msg = toProto bHash
+
+getAccountTransactionSignHashV2 :: (MonadIO m) => PreAccountTransaction -> ClientMonad m (GRPCResult (Maybe AccountTransactionSignHash))
+getAccountTransactionSignHashV2 paTransaction = withUnaryCoreV2 (callV2 @"getAccountTransactionSignHash") msg ((fmap . fmap) fromProto)
+  where
+    msg = toProto paTransaction
+
+getBlockChainParametersV2 ::
+    (MonadIO m) =>
+    BlockHashInput ->
+    ClientMonad m (GRPCResult (Maybe ChainParameterOutput))
+getBlockChainParametersV2 bHash = withUnaryCoreV2 (callV2 @"getBlockChainParameters") msg ((fmap . fmap) fromProto)
+  where
+    msg = toProto bHash
+
+getNodeInfoV2 :: (MonadIO m) => ClientMonad m (GRPCResult (Maybe NodeInfo))
+getNodeInfoV2 = withUnaryCoreV2 (callV2 @"getNodeInfo") msg ((fmap . fmap) fromProto)
+  where
+    msg = defMessage
+
+getPeersInfoV2 :: (MonadIO m) => ClientMonad m (GRPCResult (Maybe PeersInfo))
+getPeersInfoV2 = withUnaryCoreV2 (callV2 @"getPeersInfo") msg ((fmap . fmap) fromProto)
+  where
+    msg = defMessage
+
+getBlockFinalizationSummaryV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult (Maybe BlockFinalizationSummary))
+getBlockFinalizationSummaryV2 tHash = withUnaryCoreV2 (callV2 @"getBlockFinalizationSummary") msg ((fmap . fmap) fromProto)
+  where
+    msg = toProto tHash
+
+getBlockItemStatusV2 :: (MonadIO m) => TransactionHash -> ClientMonad m (GRPCResult (Maybe QueryTypes.TransactionStatus))
+getBlockItemStatusV2 tHash = withUnaryCoreV2 (callV2 @"getBlockItemStatus") msg ((fmap . fmap) fromProto)
+  where
+    msg = toProto tHash
+
+sendBlockItemV2 :: (MonadIO m) => SendBlockItemInput -> ClientMonad m (GRPCResult (Maybe TransactionHash))
+sendBlockItemV2 sbiInput = withUnaryCoreV2 (callV2 @"sendBlockItem") msg ((fmap . fmap) fromProto)
+  where
+    msg = toProto sbiInput
+
+data SendBlockItemInput
+    = AccountTransaction !Transactions.AccountTransaction
+    | AccountCreation !Transactions.AccountCreation
+    | UpdateInstruction !Updates.UpdateInstruction
+
+instance ToProto SendBlockItemInput where
+    type Output SendBlockItemInput = Proto.SendBlockItemRequest
+    toProto sbi = Proto.make $
+        case sbi of
+            AccountTransaction aTransaction ->
+                ProtoFields.accountTransaction .= toProto aTransaction
+            AccountCreation aCreation ->
+                ProtoFields.credentialDeployment .= toProto aCreation
+            UpdateInstruction uInstruction ->
+                ProtoFields.updateInstruction .= toProto uInstruction
+
+-- FIXME: Ergonomics: Should input, resp. output be ByteString?
```

Comment:
As I understand it, the `Maybe` allows you to distinguish:
- an entry that exists, but has no data (empty bytestring)
- an entry that does _not_ eixst

So I think it is fine as is. But it needs docs ;)